Server/JokesOnYou.Web.Api/Controllers/AuthController.cs

```diff
@@ -7,16 +10,53 @@
 
 namespace JokesOnYou.Web.Api.Controllers
 {
-    public class AuthController
+    [Route("api/[controller]")]
+    [ApiController]
+    public class AuthController : ControllerBase
     {
-        public ActionResult Login([FromBody] UserLoginDTO userLoginDto)
+        private readonly IAuthService _authService;
+
+        public AuthController(IAuthService authService)
         {
-            throw new NotImplementedException();
+            _authService = authService;
         }
 
-        public Task<ActionResult> Register([FromBody] UserRegisterDTO userRegisterDto)
+        [AllowAnonymous]
+        [HttpPost("login")]
+        public async Task<IActionResult> Login(UserLoginDTO userLoginDto)
         {
-            throw new NotImplementedException();
+            try
+            {
+                var user = await _authService.LoginAsync(userLoginDto);
+                return Ok(user);
+            }
+            catch (UserLoginException ex)
+            {
+                return BadRequest(ex.Message);
+            }
+            /*
+            catch (AggregateException ex)
+            {
+                return BadRequest(ex.Message);
+            }*/
+        }
+
+        [AllowAnonymous]
```

Comment:
should work without the allow anonymous since its default

Reply:
Ah yes you are right I forgot to add the [Authorize] attribute. I want the Writer of the code to be explicit in his commands so when something should be accessible to everyone it should be [AllowAnonymous]. for the case of adding a new endpoint as well.

I was also reading about DefaultIdentity There is defaults to authorize everything. that got me mixed up as well.

But either way would work, removing it or adding authorize.